.github/workflows/build.yml

```
@@ -148,7 +148,22 @@ jobs:
       run: |
         cmake -S . -B build -G Ninja -DCMAKE_BUILD_TYPE=Release
         cmake --build build
-        cpack --config build/CPackConfig.cmake -B build/installer --verbose
+        # DMG creation on macOS occasionally fails, so try multiple times
+        # https://gitlab.kitware.com/cmake/cmake/-/issues/25671
+        success=0
+        max_tries=10
+        for i in $(seq $max_tries); do
+         cmake --build build/CPackConfig.cmake -B build/installer --verbose && success=1
```

I think you only want to retry the current command multiple times?
```suggestion
         cpack --config build/CPackConfig.cmake -B build/installer --verbose && success=1
```